integrate/adaptive.py

```diff
@@ -0,0 +1,119 @@
+import heapq
+
+
+def quad_vec(f, a, b, tol, quadrature, norm_func):
+    """
+    Adaptive quadrature of vector-valued function.
+
+    Parameters
+    ----------
+    f : callable
+        Vector-valued function f(x) to integrate
+    a : float
+        Initial point.
+    b : float
+        Final point.
+    tol : float
+        Absolute tolerance.
+    quadrature : callable
+        Underlying quadrature rule to use. Should be a function like
+        quadrature(f, a, b, norm_func) which returns an estimate of the integral and
+        and an estimate of the error.
+    norm_func: callable
+        Vector norm to use for error estimation.
+
+    Returns
+    -------
+    res : {float, array-like}
```

Comment:
interface wise, this should probably always be an array, a 0D one for 1D intgrals.